include/ofi_hmem.h

```diff
@@ -37,13 +37,19 @@
 
 int cuda_copy_to_dev(void *dev, const void *host, size_t size);
 int cuda_copy_from_dev(void *host, const void *dev, size_t size);
+int cuda_query(const void *buf);
 
 static inline int ofi_memcpy(void *dest, const void *src, size_t size)
 {
 	memcpy(dest, src, size);
 	return FI_SUCCESS;
 }
 
+static inline int ofi_query(const void *buf)
```

Comment:
Let's give this a more descriptive name that indicates what is being queried or checked.  In this case, the function is trying to determine if the address is of a specific hmem type (ofi_is_hmem_iface? that would align with the higher level function name being used).

Also, I would change the 'buf' parameter to 'addr', since no size is given.  Though we could argue that a size should be provided, so that the entire range can be checked.

cuda_query and the query handler should align with the final proposal

Reply:
Will fix. In addition, I'll change to a bool.